elyra/validation/validate.py

```diff
@@ -0,0 +1,424 @@
+                                 message="Primary pipeline '{}' not found.".format(primary_pipeline_id))
+
+        if 'nodes' not in primary_pipeline or len(primary_pipeline['nodes']) == 0:
+            response.add_message(severity='1', message_type="invalidPipeline",
+                                 message="At least one node must exist in the primary pipeline.")
+        if 'app_data' not in primary_pipeline:
+            response.add_message(severity='1', message_type="invalidPipeline",
+                                 message="Pipeline 'app_data' is missing from primary pipeline")
+        elif 'version' not in primary_pipeline['app_data']:
+            response.add_message(severity='1', message_type="invalidPipeline",
+                                 message="Pipeline app_data 'version' is missing from primary pipeline")
+        elif not int(primary_pipeline['app_data']['version']) > 0:
+            response.add_message(severity='1', message_type="invalidPipeline",
+                                 message="Primary pipeline version field has an invalid value")
+
+        # TODO: Do we increment the pipeline schema version at all?
+        if pipeline_json['version'] < current_pipeline_schema_version:
+            response.add_message(severity='1', message_type="invalidPipeline",
+                                 message="Incompatible pipeline schema version detected")
+
+    def validate_compatibility(self, pipeline: Dict, response) -> None:
+        """
+        Checks that the pipeline payload is compatible with this version of elyra (ISSUE #938)
+        as well as verifying all nodes in the pipeline are supported by the runtime
+        :param pipeline: the pipeline definition to be validated
+        :param response: ValidationResponse containing the issue list to be updated
+        """
+        # TODO: Provide option to convert to current format?
+
+        pipeline_json = json.loads(json.dumps(pipeline))
+        pipeline_id = pipeline_json['pipelines'][0]['id']
+
+        # TODO: Get a list of available components from the registry, this should be done dynamically to ensure that
+        # the list of components is current and up to date since the user may have added additional ones since
+        # startup, for now use a hard-coded dict
+        supported_ops = ['execute-notebook-node']
+        if version.parse(elyra_version) >= version.parse("1.0.0"):
+            supported_ops.append('execute-python-node')
+        if version.parse(elyra_version) >= version.parse("2.2.0"):
+            supported_ops.append('execute-r-node')
+        supported_runtimes = ['kfp', 'airflow', 'null']  # need to rethink this null string
+        support_matrix = {}
+        for runtime in supported_runtimes:
+            support_matrix[runtime] = supported_ops
+
+        # Checks pipeline node types are compatible with the runtime selected
+        pipeline_runtime = pipeline_json['pipelines'][0]['app_data']['runtime']
```

Comment:
Should we add an assertion here?
```suggestion
        pipeline_runtime = pipeline_json['pipelines'][0]['app_data']['runtime']
        assert pipelne_runtime in supported_runtimes, f"Referenced runtime is not in {supported_runtimes}"
```